p2p/peer.py

```diff
@@ -853,10 +861,14 @@ def unsubscribe(self, subscriber: PeerSubscriber) -> None:
             # check, we do it here because we want to perform it for incoming peer connections as
             # well.
             msgs = await self.ensure_same_side_on_dao_fork(peer)
-        except DAOForkCheckFailure as e:
-            self.logger.debug("DAO fork check with %s failed: %s", peer, e)
+        except DAOForkCheckFailure as err:
+            self.logger.debug("DAO fork check with %s failed: %s", peer, err)
             await peer.disconnect(DisconnectReason.useless_peer)
             return
+        except MalformedMessage as err:
```

Comment:
We could instead change `ensure_same_side_on_dao_fork()` to catch `MalformedMessage` in the same place it catches `(TimeoutError, PeerConnectionLost)` (as that's closest to where the exception is actually raised) and re-raise as a `DAOForkCheckFailure`, then we don't need to do anything here

Reply:
:+1: closer to the exception source is better.